### PR TITLE
support chat-7b-8k

### PR DESCRIPTION
**Pull request by LYMDLUT**
support chat-7b-8k,otherwith it will be cut to 2048,unable to match the long sentence ability

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by ZwwWayne** [comment]
Hi @LYMDLUT ,
Thanks for your kind contribution. Simply change the context to 8k will cause the InternLM-Chat model fail to process long context window. Thus, I suggest we enhance the web_demo.py to make it allow to load different models with different configs. Would you like to do that?

**Comment by LYMDLUT** [comment]
> Hi @LYMDLUT , Thanks for your kind contribution. Simply change the context to 8k will cause the InternLM-Chat model fail to process long context window. Thus, I suggest we enhance the web_demo.py to make it allow to load different models with different configs. Would you like to do that?

Hello, Dr. Wenwei！I have made some attempts, but I am not sure if it meets your expectations. If there are any areas that need improvement, I will work hard!

File: web_demo.py [hunk 1]
```diff
@@ -199,16 +199,38 @@ def combine_history(prompt):
 
 def main():
     torch.cuda.empty_cache()
+    while True:
```
Comment (ZwwWayne) [inline web_demo.py:202]: How about we use arguments, for example 
```python
def parse_args():
    parser = argparse.ArgumentParser(description='Chat with the model')
    parser.add_argument('--model', default='InternLM-7B' help='the model type')

def main():
    args = parse_args()
```

File: web_demo.py [hunk 2]
```diff
@@ -196,19 +197,25 @@ def combine_history(prompt):
...
     print("load model begin.")
-    model, tokenizer = load_model()
+    model, tokenizer = load_model(args.model_path)
     print("load model end.")
     
     user_avator = "doc/imgs/user.png"
     robot_avator = "doc/imgs/robot.png"
     
-    st.title("InternLM-Chat-7B")
+    st.title("InternLm-7b")
```
Comment (ZwwWayne) [inline web_demo.py:216]:
```suggestion
    st.title("InternLM-Chat-7B")
```
This should be consistent with the model

File: web_demo.py [hunk 3]
```diff
@@ -196,19 +197,25 @@ def combine_history(prompt):
     total_prompt = total_prompt + cur_query_prompt.replace("{user}", prompt)
     return total_prompt
 
+def parse_args():
+    parser = argparse.ArgumentParser(description='Chat with the model, please add "--" between web_demo.py and Args in terminal')
+    parser.add_argument('--model_path', default='internlm/internlm-7b', help='path of the InternLm model')
+    parser.add_argument('--max_value', default=2048, help='the max length of the generated text')
```
Comment (ZwwWayne) [inline web_demo.py:203]:
```suggestion
    parser.add_argument('--max_value', default=2048, help='The max length of the generated text')
```

File: web_demo.py [hunk 4]
```diff
@@ -196,19 +197,25 @@ def combine_history(prompt):
     total_prompt = total_prompt + cur_query_prompt.replace("{user}", prompt)
     return total_prompt
 
+def parse_args():
+    parser = argparse.ArgumentParser(description='Chat with the model, please add "--" between web_demo.py and Args in terminal')
+    parser.add_argument('--model_path', default='internlm/internlm-7b', help='path of the InternLm model')
```
Comment (ZwwWayne) [inline web_demo.py:202]:
```suggestion
    parser.add_argument('--model', default='internlm/internlm-chat-7b', help='Type of InternLM')
```

File: README.md [hunk 5]
```diff
@@ -103,7 +103,7 @@ You can interact with the InternLM Chat 7B model through a frontend interface by
 ```bash
 pip install streamlit==1.24.0
 pip install transformers==4.30.2
-streamlit run web_demo.py
+streamlit run web_demo.py -- --model_path=internlm/internlm-chat-7b-8k --max_value=8192
```
Comment (ZwwWayne) [inline README.md:106]:
```suggestion
streamlit run web_demo.py --model_path=internlm/internlm-chat-7b-8k --max_value=8192
```

Comment (LYMDLUT) [reply README.md:106]: This is not a fault，we must add -- between web_demo.py and Args

**Comment by ZwwWayne** [comment]
This PR looks good to me now. Can be merged after resolving the conflicts.

**Comment by kkscilife** [comment]
There is no check, please commit new content to add checks before merged.

**Pull request closed without merging**